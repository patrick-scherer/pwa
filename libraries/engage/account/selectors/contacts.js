/**
 * @param {Object} state State
 * @returns {Object}
 */
const getBase = state => state.account;

/**
 * @param {Object} state State
 * @returns {Array}
 */
export const getContacts = state => getBase(state).contacts.items;
