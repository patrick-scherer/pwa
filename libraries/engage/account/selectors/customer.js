/**
 * @param {Object} state State
 * @returns {Object}
 */
const getBase = state => state.account;

/**
 * @param {Object} state State
 * @returns {Array}
 */
export const getCustomer = state => getBase(state).customer.data;
