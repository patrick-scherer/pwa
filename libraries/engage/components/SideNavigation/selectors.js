import { createSelector } from 'reselect';
import {
  getCategoryChildren,
  getRootCategories,
} from '@shopgate/pwa-common-commerce/category/selectors';

/**
 * Creates a getCategoriesById selector
 * @returns {Function}
 */
export const makeGetSubcategoriesByCategoryId = () =>
  /**
   * Retrieves categories from the state.
   * If no category id is passed, root-categories will be returned.
   * @param {Object} state The application state.
   * @param {Object} props The component props.
   * @returns {Object[]} The categories collection.
   */
  createSelector(
    getCategoryChildren,
    getRootCategories,
    (state, props) => props.categoryId,
    (childCategories, rootCategories, categoryId) => {
      // Check if we have to handle the root-category
      if (!categoryId && rootCategories) {
        return rootCategories;
      }

      return childCategories;
    }
  );

