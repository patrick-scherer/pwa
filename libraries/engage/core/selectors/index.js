export {
  getState as getShopSettingsState,
  makeGetShopSettingByKey,
  makeGetShopSettings,
  getProductImagePlaceholder,
  getCategoryImagePlaceholder,
} from './shopSettings';
export {
  getIsLocationBasedShopping,
  getFulfillmentSchedulingEnabled,
} from './merchantSettings';
