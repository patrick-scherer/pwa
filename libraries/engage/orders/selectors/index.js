import { createSelector } from 'reselect';

/**
 * Retrieves the orders state from the store.
 * @param {Object} state The current application state.
 * @return {Object} The orders state.
 */
const getOrdersState = state => state.orders || {};

/**
 * Retrieves the ordersById state from the store.
 * @param {Object} state The current application state.
 * @return {Object} The orders state.
 */
const getOrdersByIdState = state => getOrdersState(state).ordersById || {};

/**
 * Retrieves the ordersByNumber state from the store.
 * @param {Object} state The current application state.
 * @return {Object} The orders state.
 */
const getOrdersByNumberState = state => getOrdersState(state).ordersByNumber || {};

/**
 * Retrieves the orders from the store.
 * @param {Object} state The current application state.
 * @return {Object} The orders state.
 */
export const getOrders = state => getOrdersState(state).orders.orders;

/**
 * Creates the selector that retrieves an order by its number.
 * @returns {Function}
 */
export const makeGetOrderById = () => createSelector(
  (state, props) => props.orderId,
  getOrdersByIdState,
  (orderId, orders) => orders?.[orderId]?.order || null
);

/**
 * Creates the selector that retrieves an order by its number.
 * @returns {Function}
 */
export const makeGetOrderByNumber = () => createSelector(
  (state, props) => props.orderNumber,
  getOrdersByNumberState,
  (orderNumber, orders) => orders?.[orderNumber]?.order || null
);
