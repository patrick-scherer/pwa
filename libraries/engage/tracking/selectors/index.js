import { createSelector } from 'reselect';
import {
  makeGetMenu,
  makeGetIsFetchingMenu,
  makeGetShopSettingByKey,
  SHOP_SETTING_COOKIE_CONSENT_MODE,
  hasWebBridge,
} from '@shopgate/engage/core';
import { LEGAL_MENU } from '@shopgate/pwa-common/constants/MenuIDs';
import {
  COOKIE_CONSENT_MODE_OFF,
  COOKIE_CONSENT_MODE_STANDARD,
  COOKIE_CONSENT_MODE_STRICT,
} from '../constants';

/**
 * Creates a selector that retrieves the privacy polity link.
 * @returns {string}
 */
export const makeGetPrivacyPolicyLink = () => {
  const getMenu = makeGetMenu(LEGAL_MENU);
  const getIsFetchingMenu = makeGetIsFetchingMenu(LEGAL_MENU);

  return createSelector(
    getMenu,
    getIsFetchingMenu,
    (menu, fetching) => {
      if (fetching || !menu) {
        return null;
      }

      const entry = menu.find(item => item.url.includes('privacy')) || {};

      return entry?.url || null;
    }
  );
};

const getCookieConsentShopSetting = makeGetShopSettingByKey(
  SHOP_SETTING_COOKIE_CONSENT_MODE, COOKIE_CONSENT_MODE_OFF
);

export const hasCookieConsent = createSelector(
  getCookieConsentShopSetting,
  (setting) => {
    if (!hasWebBridge()) {
      return false;
    }

    return [COOKIE_CONSENT_MODE_STANDARD, COOKIE_CONSENT_MODE_STRICT].includes(setting);
  }
);

export const hasStrictCookieConsent = createSelector(
  getCookieConsentShopSetting,
  setting => setting === COOKIE_CONSENT_MODE_STRICT
);
