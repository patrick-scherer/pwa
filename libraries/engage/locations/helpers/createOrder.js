// @flow
import {
  getCurrency, getCartItems, getSubTotal, getGrandTotal,
} from '@shopgate/pwa-common-commerce/cart/selectors';
import { getProductDataById } from '@shopgate/pwa-common-commerce/product/selectors/product';
import { makeGetUserLocation } from '../selectors';

/**
 * Creates the address sequence.
 * @param {Object} formValues The reserve form values.
 * @returns {Array}
 */
function createAddressSequence(formValues) {
  return [
    {
      type: 'billing',
      firstName: formValues.firstName,
      lastName: formValues.lastName,
      phone: formValues.cellPhone,
      emailAddress: formValues.email,
    },
    {
      type: 'pickup',
      firstName: formValues.firstName2,
      lastName: formValues.lastName2,
      phone: formValues.cellPhone2,
      emailAddress: formValues.email2,
    },
  ];
}

/**
 * Retrieves the featured image for a product. If the product doesn't have a featured image
 * and is a child product, the featured image of the base product is retrieved.
 * @param {Object} product The current product.
 * @param {FUnction} getState The redux getState function.
 * @returns {string}
 */
function getProductImage(product, getState) {
  let image = product.featuredImageUrl;

  if (!image && product.baseProductId) {
    const baseProduct = getProductDataById(getState(), { productId: product.id });

    if (baseProduct) {
      image = baseProduct.featuredImageUrl;
    }
  }

  return image;
}

/**
 * Creates a single order line item (from quick reserve).
 * @param {Object} product The current product.
 * @param {Function} getState The redux getState function.
 * @returns {Array}
 */
function createSingleProductItems(product, getState) {
  const location = makeGetUserLocation()(getState());

  return [
    {
      code: product.id,
      quantity: 1,
      fulfillmentMethod: 'ROPIS', // TODO: Change based on the selected method.
      fulfillmentLocationCode: location.code,
      price: product.price.unitPrice,
      shipToAddressSequenceIndex: 0,
      currencyCode: product.price.currency,
      product: {
        code: product.id,
        name: product.name,
        image: getProductImage(product, getState),
        price: product.price.unitPrice,
        currencyCode: product.price.currency,
      },
    },
  ];
}

/**
 * Creates the order line items.
 * @param {Function} getState The redux getState function.
 * @returns {Array}
 */
function createCartLineItems(getState) {
  const items = getCartItems(getState());
  const currencyCode = getCurrency(getState());

  return items.map(item => ({
    code: item.id,
    quantity: item.quantity,
    fulfillmentMethod: item.fulfillment.method,
    fulfillmentLocationCode: item.fulfillment.location.code,
    price: item.product.price.unit,
    shipToAddressSequenceIndex: 0,
    currencyCode,
    product: {
      code: item.product.id,
      name: item.product.name,
      image: item.product.featuredImageUrl,
      price: item.product.price.unit,
      currencyCode,
    },
  }));
}

/**
 * Creates a fulfillment order out of the reserve form values and the current product or the cart.
 * @param {Object} formValues The reserve form values.
 * @param {Object} product The current product.
 * @param {Function} getState The redux getState function.
 * @returns {Object}
 */
function createOrder(formValues: { [string]: string }, product: any, getState: () => any) {
  // If no individual product was submitted, we handle the cart.
  if (product === null) {
    return {
      // localeCode: process.env.LOCALE_FILE.replace('-', '_'),
      localeCode: 'de_DE',
      currencyCode: getCurrency(getState()),
      addressSequences: createAddressSequence(formValues),
      primaryBillToAddressSequenceIndex: 0,
      lineItems: createCartLineItems(getState),
      subTotal: getSubTotal(getState()),
      total: getGrandTotal(getState()),
    };
  }

  return {
    // localeCode: process.env.LOCALE_FILE.replace('-', '_'),
    localeCode: 'de_DE',
    currencyCode: product.price.currency,
    addressSequences: createAddressSequence(formValues),
    primaryBillToAddressSequenceIndex: 0,
    lineItems: createSingleProductItems(product, getState),
    subTotal: product.price.unitPrice,
    total: product.price.unitPrice,
  };
}

export default createOrder;
