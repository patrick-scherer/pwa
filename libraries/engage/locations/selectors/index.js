// @flow
import { createSelector, type Selector } from 'reselect';
import { getProduct } from '@shopgate/engage/product';
import { getUserData } from '@shopgate/engage/user';
import { isProductAvailable } from '../helpers';
import { PRODUCT_FULFILLMENT_METHOD_DIRECT_SHIP } from '../constants';
import { type State } from '../../types';
import {
  type LocationsState,
  type LocationsByIdState,
  type LocationsByProductIdState,
  type Location,
  type UserLocationState,
  type ReservationFormValues,
  type UserLocationFulfillmentMethod,
  type UserLocationLocationCode,
} from '../locations.types';

/**
 * Retrieves the locations state from the store.
 * @param {Object} state The current application state.
 * @return {Object} The locations state.
 */
function getLocationsState(state: State): LocationsState {
  return state.locations || {};
}
/**
 * Determines the currently relevant productId from the props.
 * @param {Object} state The application state.
 * @param {Object} props The component props.
 * @param {string} props.productId The ID of a product (simple product id or parent product id)
 * @param {string} props.variantId The ID of a variant.
 * @returns {string|null}
 */
function getProductId(state, props) {
  return props.variantId || props.productId || null;
}

/**
 * Determines the currently relevant locationId from the props.
 * @param {Object} state The application state.
 * @param {Object} props The component props.
 * @param {string} props.locationId The ID of the location
 * @returns {string|null}
 */
function getLocationId(state, props) {
  return props.locationId || null;
}

/**
 * Creates the selector that retrieves the locations state.
 * @returns {Function}
 */
export function makeGetLocationsState(): Selector<State, LocationsByIdState> {
  /**
   * Retrieves the product locations state.
   * @param {Object} state The application state.
   * @returns {Object}
   */
  return createSelector(
    getLocationsState,
    state => state.locationsById || {}
  );
}

/**
 * Creates the selector that retrieves the user location state.
 * @returns {Function}
 */
export function makeGetUserLocation(): Selector<State, UserLocationState> {
  return createSelector(
    getLocationsState,
    (locations) => {
      if (!locations || !locations.userLocation) {
        return null;
      }

      return locations.userLocation;
    }
  );
}

/**
 * Creates the selector that retrieves the fulfillment method from the user location state.
 * @returns {Function}
 */
function makeGetUserLocationFulfillmentMethod(): Selector<State, UserLocationFulfillmentMethod> {
  const getUserLocation = makeGetUserLocation();
  return createSelector(
    getUserLocation,
    (userLocation) => {
      const { fulfillmentMethod = PRODUCT_FULFILLMENT_METHOD_DIRECT_SHIP } = userLocation || {};
      return fulfillmentMethod;
    }
  );
}

export { makeGetUserLocationFulfillmentMethod };

/**
 * Creates the selector that retrieves the location code from the user location state.
 * @returns {Function}
 */
export function makeGetUserLocationCode(): Selector<State, UserLocationLocationCode> {
  const getUserLocation = makeGetUserLocation();
  return createSelector(
    getUserLocation,
    (userLocation) => {
      const { code = null } = userLocation || {};
      return code;
    }
  );
}

/**
 * Creates the selector that retrieves the product locations state.
 * @returns {Function}
 */
export function makeGetProductLocationsState(): Selector<State, LocationsByProductIdState> {
  /**
   * Retrieves the product locations state.
   * @param {Object} state The application state.
   * @returns {Object}
   */
  return createSelector(
    getLocationsState,
    state => state.locationsByProductId || {}
  );
}

/**
 * Creates the selector that returns the location.
 * @returns {Function}
 */
export function makeGetLocation(): Selector<State, Location | null> {
  const getLocationsStateSelector = makeGetLocationsState();

  /**
   * Retrieves the locations.
   * @param {Object} state The application state.
   * @param {Object} props The component props.
   * @param {string} props.code The ID of the location.
   * @returns {Array|null}
   */
  return createSelector(
    getLocationsStateSelector,
    getLocationId,
    (locationsState, locationId) => {
      if (!locationId) {
        return null;
      }
      return locationsState[locationId] || null;
    }
  );
}

/**
 * Creates the selector that returns the locations for a specific product.
 * @returns {Function}
 */
export function makeGetProductLocations(): Selector<State, Location[] | null> {
  const getProductLocationsState = makeGetProductLocationsState();

  /**
   * Retrieves the locations for a specific product.
   * @param {Object} state The application state.
   * @param {Object} props The component props.
   * @param {string} props.productId The ID of the product to look for.
   * @returns {Array|null}
   */
  return createSelector(
    getProductLocationsState,
    getProductId,
    (locationsState, productId) => {
      if (!productId || !locationsState[productId]) {
        return null;
      }

      const { locations = [] } = locationsState[productId];

      return locations;
    }
  );
}

/**
 * Creates the selector that returns the single product location.
 * @param {boolean} useUserLocation Whether the location code is taken from the userLocation state.
 * @returns {Function}
 */
export function makeGetProductLocation(
  useUserLocation: boolean = false
): Selector<State, Location | null> {
  const getProductLocationsState = makeGetProductLocationsState();
  const getLocationCodeSelector = useUserLocation
    ? makeGetUserLocationCode()
    : getLocationId;

  /**
   * Retrieves the locations for a specific product.
   * @param {Object} state The application state.
   * @param {Object} props The component props.
   * @param {string} props.productId The ID of the product to look for.
   * @returns {Array|null}
   */
  return createSelector(
    getProductLocationsState,
    getLocationCodeSelector,
    getProductId,
    (locationsState, locationId, productId) => {
      if (
        !productId ||
        !locationId ||
        !locationsState[productId] ||
        !Array.isArray(locationsState[productId].locations)
      ) {
        return null;
      }

      const { locations = [] } = locationsState[productId];
      return locations.find(l => l.code === locationId) || null;
    }
  );
}

/**
 * Creates the selector that determines, if products locations for a specific product are fetching.
 * @returns {Function}
 */
export function makeGetIsFetchingProductLocations(): Selector<State, boolean | null> {
  const getProductLocationsState = makeGetProductLocationsState();

  /**
   * Retrieves the isFetching state of locations for a specific product.
   * @param {Object} state The application state.
   * @param {Object} props The component props.
   * @param {string} props.productId The ID of the product to look for.
   * @returns {Array|null}
   */
  return createSelector(
    getProductLocationsState,
    getProductId,
    (locationsState, productId) => {
      if (!productId || !locationsState[productId]) {
        return null;
      }

      const { isFetching = false } = locationsState[productId];

      return isFetching;
    }
  );
}

/**
 * Creates a selector to retrieve a product's fulfillment methods.
 * @returns {Function}
 */
export function makeGetFulfillmentMethods(): Selector<State, string[] | null> {
  /**
   * Retrieves a product's fulfillment methods.
   * @param {Object} state The application state.
   * @param {Object} props The component props.
   * @param {string} props.productId The ID of the product to look for.
   * @returns {Array|null}
   */
  return createSelector(
    getProduct,
    (product) => {
      if (!product || !product.fulfillmentMethods || product.fulfillmentMethods.length === 0) {
        return null;
      }

      return product.fulfillmentMethods;
    }
  );
}

/**
 * Creates a selector that checks if the Fulfillment Selector should be disabled.
 * @returns {Function}
 */
export function makeIsFulfillmentSelectorDisabled(): Selector<State, boolean> {
  const getProductLocations = makeGetProductLocations();
  const getFulfillmentMethods = makeGetFulfillmentMethods();

  /**
   * Retrieves whether the Fulfillment Selector should be disabled.
   * @param {Object} state The application state.
   * @param {Object} props The component props.
   * @param {string} props.productId The ID of the product to look for.
   * @returns {boolean}
   */
  return createSelector(
    getProductLocations,
    getFulfillmentMethods,
    (locations, methods) => {
      if (!methods || !locations || !locations.length === 0) {
        return true;
      }

      return false;
    }
  );
}

/**
 * Creates a selector that retrieves the user's reserve form input.
 * @returns {Function}
 */
export function makeGetUserFormInput(): Selector<State, ReservationFormValues> {
  return createSelector(
    getLocationsState,
    getUserData,
    (locations, userData) => {
      if ((!locations || !locations.userFormInput) && !userData) {
        return null;
      }
      const { firstName, lastName, mail: email } = userData || {};
      const { userFormInput = {} } = locations || {};

      return {
        firstName,
        lastName,
        email,
        ...userFormInput,
      };
    }
  );
}

/**
 * Create a selector that retrieves the user's search query input.
 * @returns {Function}
 */
export function makeGetUserSearchQuery(): Selector<State, string> {
  return createSelector(
    getLocationsState,
    locations => locations.userSearchQuery
  );
}

/**
 * Creates a selector
 * @param {boolean} useUserLocation Whether the location code is taken from the userLocation state.
 * @returns {Function}
 */
export function makeIsRopeProductOrderable(
  useUserLocation: boolean = false
): Selector<State, boolean> {
  const getUserLocationFulfillmentMethod = makeGetUserLocationFulfillmentMethod();
  const getProductLocation = makeGetProductLocation(useUserLocation);

  return createSelector(
    getUserLocationFulfillmentMethod,
    getProductLocation,
    (userLocationFulfillmentMethod, productLocation) => {
      if (
        userLocationFulfillmentMethod === PRODUCT_FULFILLMENT_METHOD_DIRECT_SHIP
      ) {
        return null;
      }

      if (productLocation === null) {
        return false;
      }

      return isProductAvailable(productLocation);
    }
  );
}
