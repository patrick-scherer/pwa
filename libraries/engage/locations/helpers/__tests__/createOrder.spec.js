import { mockedProducts } from '@shopgate/pwa-common-commerce/product/mock';
import { makeGetUserLocation } from '../../selectors';
import createOrder from '../createOrder';

jest.mock('@shopgate/engage/core');
jest.mock('@shopgate/engage/cart');
jest.mock('@shopgate/engage/product');
jest.mock('@shopgate/pwa-common/helpers/config');
jest.mock('../../selectors', () => ({
  makeGetUserLocation: jest.fn(),
}));

describe('libraries > engage > locations > helpers > createOrder', () => {
  const getState = jest.fn();
  const formValues = {
    firstName: 'firstName',
    lastName: 'lastName',
    cellPhone: 'cellPhone',
    email: 'email',
    firstName2: 'firstName2',
    lastName2: 'lastName2',
    cellPhone2: 'cellPhone2',
    email2: 'email2',
  };

  it('should create order for single product', () => {
    makeGetUserLocation.mockImplementation(() => jest.fn().mockReturnValue({
      code: 'LOCATION_CODE',
    }));
    getState.mockReturnValueOnce({ user: { data: {} } });
    expect(createOrder(formValues, mockedProducts.products[0], getState)).toMatchSnapshot();

    getState.mockReturnValueOnce({ user: { data: { id: 123456 } } });
    expect(createOrder(formValues, mockedProducts.products[0], getState)).toMatchSnapshot();
  });

  it('should create order for cart items', () => {
    getState.mockReturnValueOnce({ user: { data: {} } });
    expect(createOrder(formValues, null, getState)).toMatchSnapshot();

    getState.mockReturnValueOnce({ user: { data: { id: 123456 } } });
    expect(createOrder(formValues, null, getState)).toMatchSnapshot();
  });
});
