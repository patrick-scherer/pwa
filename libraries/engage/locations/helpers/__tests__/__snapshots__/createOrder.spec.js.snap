// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`libraries > engage > locations > helpers > createOrder should create order for cart items 1`] = `
Object {
  "addressSequences": Array [
    Object {
      "country": "DE",
      "emailAddress": "email",
      "firstName": "firstName",
      "lastName": "lastName",
      "phone": "cellPhone",
      "type": "billing",
    },
    Object {
      "country": "DE",
      "emailAddress": "email2",
      "firstName": "firstName2",
      "lastName": "lastName2",
      "phone": "cellPhone2",
      "type": "pickup",
    },
  ],
  "currencyCode": "EUR",
  "externalCustomerNumber": undefined,
  "lineItems": Array [
    Object {
      "code": "106051878",
      "currencyCode": "EUR",
      "fulfillmentLocationCode": "LOCATION_CODE",
      "fulfillmentMethod": "ROPIS",
      "price": 200,
      "product": Object {
        "code": "2016",
        "currencyCode": "EUR",
        "image": "https://picsum.photos/440/440",
        "name": "Honda Accord 2016 Sedan",
        "options": Array [
          Object {
            "code": "color",
            "name": "Color",
            "value": Object {
              "code": "black",
              "name": "Black",
            },
          },
          Object {
            "code": "shoe_size",
            "name": "Shoe size",
            "value": Object {
              "code": "5",
              "name": "5",
            },
          },
        ],
        "price": 100,
      },
      "quantity": 2,
      "shipToAddressSequenceIndex": 1,
    },
  ],
  "localeCode": "de-de",
  "os": "android",
  "platform": "engage",
  "primaryBillToAddressSequenceIndex": 0,
  "subTotal": 200,
  "total": 200,
  "userAgent": "Mozilla/5.0 (Linux; Android 8.0.0; Pixel 2 XL Build/OPD1.170816.004) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Mobile Safari/537.36",
}
`;

exports[`libraries > engage > locations > helpers > createOrder should create order for cart items 2`] = `
Object {
  "addressSequences": Array [
    Object {
      "country": "DE",
      "emailAddress": "email",
      "firstName": "firstName",
      "lastName": "lastName",
      "phone": "cellPhone",
      "type": "billing",
    },
    Object {
      "country": "DE",
      "emailAddress": "email2",
      "firstName": "firstName2",
      "lastName": "lastName2",
      "phone": "cellPhone2",
      "type": "pickup",
    },
  ],
  "currencyCode": "EUR",
  "externalCustomerNumber": "123456",
  "lineItems": Array [
    Object {
      "code": "106051878",
      "currencyCode": "EUR",
      "fulfillmentLocationCode": "LOCATION_CODE",
      "fulfillmentMethod": "ROPIS",
      "price": 200,
      "product": Object {
        "code": "2016",
        "currencyCode": "EUR",
        "image": "https://picsum.photos/440/440",
        "name": "Honda Accord 2016 Sedan",
        "options": Array [
          Object {
            "code": "color",
            "name": "Color",
            "value": Object {
              "code": "black",
              "name": "Black",
            },
          },
          Object {
            "code": "shoe_size",
            "name": "Shoe size",
            "value": Object {
              "code": "5",
              "name": "5",
            },
          },
        ],
        "price": 100,
      },
      "quantity": 2,
      "shipToAddressSequenceIndex": 1,
    },
  ],
  "localeCode": "de-de",
  "os": "android",
  "platform": "engage",
  "primaryBillToAddressSequenceIndex": 0,
  "subTotal": 200,
  "total": 200,
  "userAgent": "Mozilla/5.0 (Linux; Android 8.0.0; Pixel 2 XL Build/OPD1.170816.004) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Mobile Safari/537.36",
}
`;

exports[`libraries > engage > locations > helpers > createOrder should create order for single product 1`] = `
Object {
  "addressSequences": Array [
    Object {
      "country": "DE",
      "emailAddress": "email",
      "firstName": "firstName",
      "lastName": "lastName",
      "phone": "cellPhone",
      "type": "billing",
    },
    Object {
      "country": "DE",
      "emailAddress": "email2",
      "firstName": "firstName2",
      "lastName": "lastName2",
      "phone": "cellPhone2",
      "type": "pickup",
    },
  ],
  "currencyCode": "EUR",
  "externalCustomerNumber": undefined,
  "lineItems": Array [
    Object {
      "code": "SG117",
      "currencyCode": "EUR",
      "fulfillmentLocationCode": "LOCATION_CODE",
      "fulfillmentMethod": "ROPIS",
      "price": 100,
      "product": Object {
        "code": "SG117",
        "currencyCode": "EUR",
        "image": null,
        "name": "no_description",
        "options": Array [
          Object {
            "code": "1",
            "name": "Color",
            "value": Object {
              "code": "black",
              "name": "Black",
            },
          },
          Object {
            "code": "2",
            "name": "Shoe size",
            "value": Object {
              "code": "5",
              "name": "5",
            },
          },
        ],
        "price": 100,
      },
      "quantity": 1,
      "shipToAddressSequenceIndex": 1,
    },
  ],
  "localeCode": "de-de",
  "os": "android",
  "platform": "engage",
  "primaryBillToAddressSequenceIndex": 0,
  "subTotal": 100,
  "total": 100,
  "userAgent": "Mozilla/5.0 (Linux; Android 8.0.0; Pixel 2 XL Build/OPD1.170816.004) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Mobile Safari/537.36",
}
`;

exports[`libraries > engage > locations > helpers > createOrder should create order for single product 2`] = `
Object {
  "addressSequences": Array [
    Object {
      "country": "DE",
      "emailAddress": "email",
      "firstName": "firstName",
      "lastName": "lastName",
      "phone": "cellPhone",
      "type": "billing",
    },
    Object {
      "country": "DE",
      "emailAddress": "email2",
      "firstName": "firstName2",
      "lastName": "lastName2",
      "phone": "cellPhone2",
      "type": "pickup",
    },
  ],
  "currencyCode": "EUR",
  "externalCustomerNumber": "123456",
  "lineItems": Array [
    Object {
      "code": "SG117",
      "currencyCode": "EUR",
      "fulfillmentLocationCode": "LOCATION_CODE",
      "fulfillmentMethod": "ROPIS",
      "price": 100,
      "product": Object {
        "code": "SG117",
        "currencyCode": "EUR",
        "image": null,
        "name": "no_description",
        "options": Array [
          Object {
            "code": "1",
            "name": "Color",
            "value": Object {
              "code": "black",
              "name": "Black",
            },
          },
          Object {
            "code": "2",
            "name": "Shoe size",
            "value": Object {
              "code": "5",
              "name": "5",
            },
          },
        ],
        "price": 100,
      },
      "quantity": 1,
      "shipToAddressSequenceIndex": 1,
    },
  ],
  "localeCode": "de-de",
  "os": "android",
  "platform": "engage",
  "primaryBillToAddressSequenceIndex": 0,
  "subTotal": 100,
  "total": 100,
  "userAgent": "Mozilla/5.0 (Linux; Android 8.0.0; Pixel 2 XL Build/OPD1.170816.004) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Mobile Safari/537.36",
}
`;
